src/ir/LocalGraph.cpp

```diff
@@ -52,6 +52,13 @@ struct Flower : public CFGWalker<Flower, Visitor<Flower>, Info> {
 
   BasicBlock* makeBasicBlock() { return new BasicBlock(); }
 
+  void continueToNewBasicBlock() {
```

Comment:
Are there other reasonable things this could be overridden with? I'm not sure there are, and if there were, they would require updating the internal state of the `CFGWalker`, which seems less than ideal. Maybe this could all be simplified if CFGWalker just took a bool (or similar enum) template parameter to configure this behavior?

Reply:
A bool would work here, yeah. But in general another option here could be to check if the block is empty and merge it in that case. That is, if there are no other instructions a pass cares about, then it might be ok with merging the continuation, but not otherwise. That would require inspecting `block->contents` etc. So overall I think a hook like this makes sense.

Reply:
(As a datapoint, I saw that 15% of blocks in LocalGraph are empty, that is, contain no LocalGet or Set instructions. That number is even higher for other passes, I'd bet, that focus on less common instructions. So there could be nice optimization opportunities here.)